src/CALab/Grid.java

```diff
@@ -81,14 +119,33 @@ public Set<Cell> getNeighbors(Cell asker, int radius) {
 
     public void observe() {
         // call each cell's observe method and notify subscribers
+        for(int row = 0; row < cells.length; row++) {
+            for (int col = 0; col < cells[row].length; col++) {
+                Cell cell = getCell(row, col);
+                cell.observe();
+                cell.notifySubscribers();
+            }
+        }
```

Comment:
this can/should be simplified to `for (Cell cell : cells[row]) {` and then you can delete line 124; also i believe we need to have `notifySubscribers()` at the end of this method. See suggestion below:
```suggestion
        for(int row = 0; row < cells.length; row++) {
            for (Cell cell : cells[row]) {
                cell.observe();
                cell.notifySubscribers();
            }
        }
        notifySubscribers();
```